File: scripts/graphrag/rebuild_store.js
const fs = require('fs');
const path = require('path');
const http = require('http');
const jwt = require('jsonwebtoken');

const token = jwt.sign({ sub: 'rebuild' }, process.env.JWT_SECRET, { expiresIn: '1h' });
const docsDir = '/tmp/graphrag-test';
const PORT = Number(process.env.GRAPH_RAG_PORT || 8001);

const files = {
  'semaa-policy-001': '01_marine_policy.txt',
  'semaa-incident-003': '02_incident_report.txt',
  'semaa-contract-088': '03_supplier_contract.txt',
  'semaa-portlog-014': '04_port_maintenance_log.txt',
  'semaa-training-q1': '05_crew_training_record.txt',
  'semaa-budget-h1': '06_budget_and_procurement_plan.txt',
};

function ingest(file_id, filename, text) {
  return new Promise((resolve, reject) => {
    const body = JSON.stringify({ file_id, filename, text });
    const opts = {
      hostname: 'localhost',
      port: PORT,
      path: '/ingest-text',
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': 'Bearer ' + token,
        'Content-Length': Buffer.byteLength(body),
      },
    };
    const req = http.request(opts, (res) => {
      let d = '';
      res.on('data', (c) => { d += c; });
      res.on('end', () => resolve(JSON.parse(d)));
    });
    req.on('error', reject);
    req.write(body);
    req.end();
  });
}

(async () => {
  console.log('GraphRAG Store Rebuild');
  console.log('='.repeat(40));
  for (const [file_id, filename] of Object.entries(files)) {
    const filepath = path.join(docsDir, filename);
    if (!fs.existsSync(filepath)) {
      console.log('SKIP ' + filename + ' — not found');
      continue;
    }
    const text = fs.readFileSync(filepath, 'utf8');
    const r = await ingest(file_id, filename, text);
    console.log(filename + ': chunks=' + r.chunk_count + ', entities=' + r.entity_count + ', ok=' + r.status);
  }
  console.log('='.repeat(40));
  console.log('Done.');
})();

File: scripts/graphrag/rebuild_store.py
#!/usr/bin/env python3
"""Rebuild GraphRAG store by re-ingesting all documents."""
import json
import os
import subprocess
import urllib.request

BASE_DIR = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
DOCS_DIR = os.path.join(BASE_DIR, "docs", "graphrag-test")
GRAPHRAG_URL = "http://localhost:8001"

FILES = {
    "semaa-policy-001": "01_marine_policy.txt",
    "semaa-incident-003": "02_incident_report.txt",
    "semaa-contract-088": "03_supplier_contract.txt",
    "semaa-portlog-014": "04_port_maintenance_log.txt",
    "semaa-training-q1": "05_crew_training_record.txt",
    "semaa-budget-h1": "06_budget_and_procurement_plan.txt",
}

def get_token():
    result = subprocess.check_output([
        "docker", "exec", "graphrag_api", "node", "-e",
        'const jwt=require("jsonwebtoken"); '
        'console.log(jwt.sign({sub:"rebuild"}, process.env.JWT_SECRET, {expiresIn:"1h"}));'
    ])
    return result.decode().strip()

def ingest(file_id, filename, text, token):
    payload = json.dumps({"file_id": file_id, "filename": filename, "text": text}).encode()
    req = urllib.request.Request(
        f"{GRAPHRAG_URL}/ingest-text",
        data=payload,
        headers={"Content-Type": "application/json", "Authorization": f"Bearer {token}"},
        method="POST",
    )
    with urllib.request.urlopen(req, timeout=30) as resp:
        return json.loads(resp.read())

def main():
    print("GraphRAG Store Rebuild")
    print("=" * 40)
    token = get_token()
    for file_id, filename in FILES.items():
        path = os.path.join(DOCS_DIR, filename)
        if not os.path.exists(path):
            print(f"  SKIP {filename} — not found")
            continue
        with open(path, encoding="utf-8") as f:
            text = f.read()
        result = ingest(file_id, filename, text, token)
        print(f"  {filename}: chunks={result.get('chunk_count')}, entities={result.get('entity_count')}, ok={result.get('status')}")
    print("=" * 40)
    print("Done.")

if __name__ == "__main__":
    main()
